ext/SummationByPartsOperatorsOptimForwardDiffExt.jl

```diff
@@ -0,0 +1,222 @@
+module SummationByPartsOperatorsOptimForwardDiffExt
+
+using Optim: Optim, Options, LBFGS, optimize, minimizer
+using ForwardDiff: ForwardDiff
+
+using SummationByPartsOperators: SummationByPartsOperators, GlaubitzNordströmÖffner2023, MatrixDerivativeOperator
+using LinearAlgebra: Diagonal, LowerTriangular, dot, diag, norm, mul!
+using PreallocationTools: DiffCache, get_tmp
+using SparseArrays: spzeros
+
+function SummationByPartsOperators.function_space_operator(basis_functions,
+                                                           x_min::T, x_max::T, nodes::Vector{T},
```

Comment:
I'm also not sure how to deal with `x_min` and `x_max`. We could

1. Keep it as it is, which means that if `x_min` and/or `x_max` are not within `nodes`, then the grid of the `MatrixDerivativeOperator` is not the same as the `nodes`, which feels unintuitive, when you think of `D` as global operator and not an operator on a reference element.
2. Error if `x_min` and/or `x_max` are not within `nodes`.
3. If they are not within `nodes` implicitly add them, which also doesn't feel quite correct.
4. Remove the arguments `x_min` and `x_max` and just pass `first(nodes)` and `last(nodes)` to the `MatrixDerivativeOperator`.

I think I tend towards option 4. What is your opinion, @ranocha?

Reply:
Yes, that sounds reasonable. We should mention this in the docstring.

Reply:
Thanks!